dts: don't use 'test' vendor prefix

mbolivar-nordic opened the pull request:
This is now failing an edtlib check for unknown vendor prefixes.

I can't find a reason to use a vendor prefix in application-local
bindings like this, so just remove it wherever it appears by
normalizing to test-foo-compat instead of test,foo_compat or
test,foo-compat.

mbolivar-nordic commented:
Fixes CI issues from the latest daily build pointed out in `#infrastructure` on slack.

mniestroj commented on tests/drivers/regulator/fixed/dts/bindings/regulator-fixed.yaml:
```diff
@@ -5,7 +5,7 @@ description: |
     tests/drivers/regulator/fixed test in Zephyr.
 
-compatible: "test,regulator-fixed"
+compatible: "regulator-fixed"
```
Doesn't it conflict with https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/regulator/regulator-fixed.yaml ?

mbolivar-nordic replied:
Yep, normalizing a different way now.

mbolivar-nordic committed:
dts: don't use 'test' vendor prefix

This is now failing an edtlib check for unknown vendor prefixes.

I can't find a reason to use a vendor prefix in application-local
bindings like this, so just remove it wherever it appears by
normalizing to test-foo-compat instead of test,foo_compat or
test,foo-compat.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>
